lib/MySQL_Thread.cpp

```diff
@@ -1464,3 +1872,30 @@ void MySQL_Threads_Handler::signal_all_threads() {
 	}
 }
 
+bool MySQL_Threads_Handler::kill_session(uint32_t _thread_session_id) {
+	bool ret=false;
+	unsigned int i;
+	signal_all_threads(1);
+	for (i=0;i<num_threads;i++) {
```

Comment:
I see. I didn't spot the usage in MySQL_Thread::run. Thanks for the explanation!